internal/internal_worker.go

```diff
@@ -874,66 +875,51 @@ func (aw *aggregatedWorker) Start() error {
 	return nil
 }
 
-var binaryChecksum string
-var binaryChecksumLock sync.Mutex
+var binaryChecksum atomic.Value
 
 // SetBinaryChecksum set binary checksum
 func SetBinaryChecksum(checksum string) {
-	binaryChecksumLock.Lock()
-	defer binaryChecksumLock.Unlock()
-
-	binaryChecksum = checksum
-}
-
-func initBinaryChecksum() error {
-	binaryChecksumLock.Lock()
-	defer binaryChecksumLock.Unlock()
-
-	return initBinaryChecksumLocked()
+	binaryChecksum.Store(checksum)
 }
 
-// callers MUST hold binaryChecksumLock before calling
-func initBinaryChecksumLocked() error {
-	if len(binaryChecksum) > 0 {
-		return nil
-	}
-
+func initBinaryChecksum() (string, error) {
 	exec, err := os.Executable()
```

Comment:
still needs this check - sequential startup has no need to re-compute the same thing repeatedly, and it'll also clobber whatever your current manually-set checksum might be.